src/v2i-hub/RSUHealthMonitorPlugin/src/RSUHealthMonitorPlugin.cpp

```diff
@@ -0,0 +1,220 @@
+        GetConfigValue<string>("AuthPassPhrase", _authPassPhrase);
+        GetConfigValue<string>("SecurityUser", _securityUser);
+
+        // Update the OID to RSU field mapping
+        string rsuOIDMapJsonStr;
+        GetConfigValue<string>("RSUOIDConfigMap", rsuOIDMapJsonStr);
+        UpdateRSUOIDConfig(rsuOIDMapJsonStr);
+    }
+
+    void RSUHealthMonitorPlugin::UpdateRSUOIDConfig(string &json_str)
+    {
+
+        if (json_str.length() == 0)
+        {
+            PLOG(logERROR) << "Error updating RSU OID config due to JSON is empty.";
+            return;
+        }
+        try
+        {
+            ptree pt;
+            istringstream iss(json_str);
+            read_json(iss, pt);
+
+            // Clear the RSU OID mapping variable
+            _rsuOIDConfigMap.clear();
+            BOOST_FOREACH (ptree::value_type &child, pt.get_child("RSUOIDConfig"))
+            {
+                // Array elements have no names.
+                assert(child.first.empty());
+                RSUOIDConfig config;
+                config.field = child.second.get<string>("RsuField");
+                config.oid = child.second.get<string>("OID");
+                config.required = child.second.get<bool>("Required");
+                // Add RSU OID to the map
+                _rsuOIDConfigMap.push_back(config);
+            }
+        }
+        catch (const std::exception &e)
+        {
+            PLOG(logERROR) << "Error updating RSU OID config" << e.what();
+        }
+    }
+
+    void RSUHealthMonitorPlugin::OnConfigChanged(const char *key, const char *value)
+    {
+        PluginClient::OnConfigChanged(key, value);
+        UpdateConfigSettings();
+    }
+
+    void RSUHealthMonitorPlugin::PeriodicRSUStatusReq()
+    {
+        while (true)
+        {
+            // Broadcast the RSU status info when there are any RSU responses.
+            try
+            {
+                auto rsuStatusJson = getRSUstatus();
+
+                if (!rsuStatusJson.empty())
+                {
+                    vector<string> rsuStatusFields;
+                    for (auto const &field : rsuStatusJson.getMemberNames())
+                    {
+                        rsuStatusFields.push_back(field);
+                    }
+                    // Only broadcast RSU status when all required fields are present.
+                    if (isAllRequiredFieldsPresent(rsuStatusFields))
+                    {
+                        Json::FastWriter fasterWirter;
+                        string json_str = fasterWirter.write(rsuStatusJson);
+                        tmx::messages::RSUStatusMessage sendRsuStatusMsg;
+                        sendRsuStatusMsg.set_contents(json_str);
+                        BroadcastMessage(sendRsuStatusMsg);
+                        PLOG(logINFO) << "Broadcast RSU status:  " << json_str;
+                    }
+                }
+            }
+            catch (std::exception &ex)
```

Comment:
This should be const

Reply:
Addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/564/commits/c0bac101e0cef01c154929024be6ab01b263785b